pyomo/core/plugins/transform/discrete_vars.py

```diff
@@ -25,56 +41,170 @@
     'core.relax_integer_vars', doc="Relax integer variables to continuous counterparts"
 )
 class RelaxIntegerVars(Transformation):
+    CONFIG = ConfigDict('core.relax_integer_vars')
+    CONFIG.declare(
+        'targets',
+        ConfigValue(
+            default=None,
+            domain=target_list,
+            description="target or list of targets that will be relaxed",
+            doc="""
+            This specifies the list of components to relax. If None (default), the
+            entire model is transformed. Note that if the transformation is done
+            out of place, the list of targets should be attached to the model before
+            it is cloned, and the list will specify the targets on the cloned
+            instance.""",
+        ),
+    )
+    CONFIG.declare(
+        'reverse',
+        ConfigValue(
+            default=None,
+            description="The token returned by a (forward) call to this "
+            "transformation, if you wish to reverse the transformation.",
+            doc="""
+            This argument should be the reverse transformation token
+            returned by a previous call to this transformation to transform
+            fixed disjunctive state in the given model.
+            If this argument is specified, this call to the transformation
+            will reverse what the transformation did in the call that returned
+            the token. Note that if there are intermediate changes to the model
+            in between the forward and the backward calls to the transformation,
+            the behavior could be unexpected.
+            """,
+        ),
+    )
+    CONFIG.declare(
+        'var_collector',
+        ConfigValue(
+            default=VarCollector.FromVarComponents,
+            domain=In(VarCollector),
+            description="The method for collection the Vars to relax. If "
+            "VarCollector.FromVarComponents (default), any Var component on "
+            "the active tree will be relaxed.",
+            doc="""
+            This specifies the method for collecting the Var components to relax.
+            The default, VarCollector.FromVarComponents, assumes that all relevant
+            Vars are on the active tree passed to the transformation. If this is
```

Comment:
Something with this grammar is weird. I'm not sure what needs to be fixed, but it's not quite right.